### PR TITLE
Fix missing download button for old layout

### PR DESCRIPTION
**Pull request by jorgenherje**
Download button not appearing for old plugin layout implementations.

- Fix breaking change
- Remove usage of unused prop from https://github.com/equinor/webviz-core-components/pull/232
- Ensure download button appearing and download callback being called for old layout implementation

----

Dependent on https://github.com/equinor/webviz-core-components/pull/232

### CODE REVIEW

File: webviz_config/_plugin_abc.py [hunk 1]
```diff
@@ -332,14 +325,18 @@ def get_all_settings(self) -> List[html.Div]:
     def _plugin_wrapper_id(self) -> str:
         return f"plugin-wrapper-{self._plugin_unique_id}"
 
+    @property
+    def _legacy_plugin_view_id(self) -> str:
+        return f"{self._plugin_unique_id.to_string()}-view"
```
Comment (anders-kiaer) [inline webviz_config/_plugin_abc.py:330]: The `@property` (indicating a public property) and the leading`_` (indicating private attribute) is a bit conflicting perhaps?

File: webviz_config/_plugin_abc.py [hunk 2]
```diff
@@ -86,13 +86,6 @@ def layout(self):
     # E.g. download of zip archive will only appear if the plugin also
     # has defined the corresponding callback, and contact person will only
     # appear if the user configuration file has this information.
-    TOOLBAR_BUTTONS = [
-        "screenshot",
-        "expand",
-        "contact_person",
-        "guided_tour",
-        "feedback",
-    ]
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply webviz_config/_plugin_abc.py:95]: Sounds fair - have removed references and updated the documentation/description

File: webviz_config/_plugin_abc.py (hunk 1 above)
Comment (jorgenherje) [reply webviz_config/_plugin_abc.py]: Converted the property to a private attribute in `__init__`

File: webviz_config/_plugin_abc.py [hunk 3]
```diff
@@ -456,33 +451,26 @@ def plugin_layout(
...
-        the plugin layout wrapped into a common webviz config plugin
-        component, which provides some useful buttons like download of data,
-        show data contact person and download plugin content to png.
-
-        CSV download button will only appear if the plugin class has a property
-        `csv_string` which should return the appropriate csv data as a string.
-
-        If `TOOLBAR_BUTTONS` is empty, this functions returns the same
-        dash layout as the plugin class provides directly.
+        """This function returns plugin layour placed within a WebvizPluginWrapper
```
Comment (anders-kiaer) [inline webviz_config/_plugin_abc.py:454]:
```suggestion
        """This function returns plugin layout placed within a WebvizPluginWrapper
```

**Review by anders-kiaer — APPROVED** [review]

**Pull request merged**